ADR/0058-maven-build-pipeline.md

```diff
@@ -0,0 +1,143 @@
+# 58. Maven Build Pipelines
+
+Date: 2025-12-08
+
+## Status
+
+Implementable
+
+## Context
+
+Currently, Konflux lacks a native Java build experience, forcing developers to use Containerfiles
+for Java builds. The results are pushed as container images, which cannot be easily consumed by
+dependent Java projects.
+
+Apache [Maven](https://maven.apache.org/index.html) is the most widely adopted build tool for Java
+projects, although it is not hold and exclusive position. Java projects with Maven are distributed
+through [Maven Repositories](https://maven.apache.org/repositories/index.html), whose model has
+adopted by other Java build tools like [Gradle](https://gradle.org/).
+
+### Goals for this feature
+
+- Provide a primary build experience for Java projects using Maven within Konflux.
+- Enable developers to specify custom Maven commands ("goals") during the build process, as well as
+  input parameters/arguments.
+- Facilitate the publication of Java artifacts (JAR, WAR, POM) to a configured Maven repository for
+  use in a wider Konflux Application (or `ComponentGroup`).
+
+
+### Key Requirements
+
+- Developers must be able to select a base builder image containing a desired OpenJDK and Maven
+  version. Red Hat provides OpenJDK builders bundled with Maven 3.x through its UBI8 and UBI9
+  catalogs.
+- The pipeline must invoke a reasonable default set of `mvn` commands (`mvn clean deploy`) to a
+  "dev" repository when code is merged. Code should not be deployed to a Maven repository for pull
+  request builds.
+- Developers must be able to provide arguments and CLI flags to the `mvn` command.
+- The build must produce OCI artifact references (`*_IMAGE_URL` and `*_IMAGE_DIGEST` or `IMAGES`)
+  for Tekton Chains to generate provenance and signatures.
+- Support for [multi-module](https://maven.apache.org/guides/mini/guide-multiple-modules.html)
+  projects is required.
+
+### Non-Goals
+
+- Support for [Maven 4](https://maven.apache.org/whatsnewinmaven4.html).
+- Support for Gradle.
+- Support for Kotlin, Scala, or other JVM-based programming languages.
+- Provision Maven repositories for Java projects.
+
+These can all be addressed through follow-up ADRs.
+
+## Decision
+
+The build pipeline will execute up to two "deployments":
```

Comment:
I think "deployments" here is a confusing terminology. Can we reword in terms of "pushing content"? That's a bit less ambiguous.